### PR TITLE
Add Dirichlet boundary condition for CurvedScalarWave

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

I propose to add this simple boundary condition (BC) for CurvedScalarWave (CSW) in order to be able to write a fully Ghost type boundary condition for the ScalarTensor system (#5235). This is because they are the simplest conditions to implement in the combined system and also because the CSW constraint preserving BCs may not trivially generalize to the ScalarTensor system (where the metric is evolving as well).

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/AnalyticConstant.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  BoundaryCondition<Dim>::pup(p);
+  p | amplitude_;
+}
+
+template <size_t Dim>
+AnalyticConstant<Dim>::AnalyticConstant(CkMigrateMessage* const msg)
+    : BoundaryCondition<Dim>(msg) {}
+
+template <size_t Dim>
+AnalyticConstant<Dim>::AnalyticConstant(const double amplitude) {
```
Comment (nilsdeppe) [inline src/Evolution/Systems/CurvedScalarWave/BoundaryConditions/AnalyticConstant.cpp]: You should prefer list initialization whenever possible:
```cpp
AnalyticConstant<Dim>::AnalyticConstant(const double amplitude) : amplitude_(amplitude) {}
```

File: tests/Unit/Evolution/Systems/CurvedScalarWave/BoundaryConditions/Test_AnalyticConstant.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          helpers::Tags::PythonFunctionName<
+              CurvedScalarWave::Tags::ConstraintGamma1>,
+          helpers::Tags::PythonFunctionName<
+              CurvedScalarWave::Tags::ConstraintGamma2>>{
+          "error", "psi_analytic_constant", "pi_analytic_constant",
+          "phi_analytic_constant", "lapse_analytic_constant",
+          "shift_analytic_constant", "constraint_gamma1_analytic_constant",
+          "constraint_gamma2_analytic_constant"},
+      "AnalyticConstant:\n"
+      "  Amplitude: 0.0\n",
```
Comment (nilsdeppe) [inline tests/Unit/Evolution/Systems/CurvedScalarWave/BoundaryConditions/Test_AnalyticConstant.cpp]: what about setting the Amplitude to some non-zero & non-1 number, e.g. `2.3` and then just hard-code the same number in the python file

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, please squash the minor suggestions immediately

**Comment by guilara** [comment]
@nilsdeppe Thanks for the review. I've implemented the suggestions and rebased

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**